torchaudio/__init__.py

```diff
@@ -125,85 +148,26 @@ def save(filepath, src, sample_rate, precision=16, channels_first=True):
-            audio type cannot be automatically determined. (Default: ``None``)
-        encodinginfo (sox_encodinginfo_t, optional): A sox_encodinginfo_t type, which could be set if the
-            audio type cannot be automatically determined. (Default: ``None``)
-        filetype (str, optional): A filetype or extension to be set if sox cannot determine it
-            automatically. (Default: ``None``)
+        return sox_backend.save(
+            filepath, src, sample_rate, precision, channels_first=channels_first
+        )
+    elif get_audio_backend() == "soundfile":
+        from torchaudio import _soundfile_backend
 
-    Example
-        >>> data, sample_rate = torchaudio.load('foo.mp3')
-        >>> torchaudio.save('foo.wav', data, sample_rate)
+        return _soundfile_backend.save(filepath, src, sample_rate, channels_first)
+    else:
+        raise NotImplementedError
 
-    """
-    ch_idx, len_idx = (0, 1) if channels_first else (1, 0)
-
-    # check if save directory exists
-    abs_dirpath = os.path.dirname(os.path.abspath(filepath))
-    if not os.path.isdir(abs_dirpath):
-        raise OSError("Directory does not exist: {}".format(abs_dirpath))
-    # check that src is a CPU tensor
-    check_input(src)
-    # Check/Fix shape of source data
-    if src.dim() == 1:
-        # 1d tensors as assumed to be mono signals
-        src.unsqueeze_(ch_idx)
-    elif src.dim() > 2 or src.size(ch_idx) > 16:
-        # assumes num_channels < 16
-        raise ValueError(
-            "Expected format where C < 16, but found {}".format(src.size()))
-    # sox stores the sample rate as a float, though practically sample rates are almost always integers
-    # convert integers to floats
-    if not isinstance(signalinfo.rate, float):
-        if float(signalinfo.rate) == signalinfo.rate:
-            signalinfo.rate = float(signalinfo.rate)
-        else:
-            raise TypeError('Sample rate should be a float or int')
-    # check if the bit precision (i.e. bits per sample) is an integer
-    if not isinstance(signalinfo.precision, int):
-        if int(signalinfo.precision) == signalinfo.precision:
-            signalinfo.precision = int(signalinfo.precision)
-        else:
-            raise TypeError('Bit precision should be an integer')
-    # programs such as librosa normalize the signal, unnormalize if detected
-    if src.min() >= -1.0 and src.max() <= 1.0:
-        src = src * (1 << 31)
-        src = src.long()
-    # set filetype and allow for files with no extensions
-    extension = os.path.splitext(filepath)[1]
-    filetype = extension[1:] if len(extension) > 0 else filetype
-    # transpose from C x L -> L x C
-    if channels_first:
-        src = src.transpose(1, 0)
-    # save data to file
-    src = src.contiguous()
-    _torch_sox.write_audio_file(filepath, src, signalinfo, encodinginfo, filetype)
+
+def save_encinfo(*args, **kwargs):
```

Comment:
The [current docs for encinfo](https://github.com/pytorch/audio/blob/c74e580f92c11419a9a3c02fd2d0e301e3347597/torchaudio/__init__.py#L140) don't even reference the function in the example. This is a strange one.